File: java/java-sdk/java-sdk-test/src/test/java/io/kestra/example/BasicSDKUsageExampleTest.java
package io.kestra.example;

import io.kestra.sdk.internal.ApiException;
import org.junit.jupiter.api.Test;

import static io.kestra.TestUtils.TENANT;
import static io.kestra.TestUtils.client;
import static org.assertj.core.api.Assertions.assertThat;

/**
 * Runs the curated README example end-to-end against the live Kestra container.
 *
 * <p>This is the CI guarantee for the injected snippet: if the code in
 * {@link BasicSDKUsageExample} (and therefore the README block produced from it
 * by {@code test-utils/embed_snippets.py}) ever stops matching the SDK, this
 * test fails. See issue #144.
 */
class BasicSDKUsageExampleTest {

    @Test
    void flowLifecycle_runsEndToEnd() throws ApiException {
        try {
            String executionId = BasicSDKUsageExample.flowLifecycle(client(), TENANT);
            assertThat(executionId).isNotBlank();
        } finally {
            // Best-effort cleanup so the example is re-runnable within a session.
            try {
                client().flows().deleteFlow("company.team", "hello_from_sdk", TENANT);
            } catch (ApiException ignored) {
                // flow may not exist if the example failed before creating it
            }
        }
    }
}

File: go-sdk/test/basic_sdk_usage_example_test.go
package test

import (
	"context"
	"testing"

	"github.com/stretchr/testify/require"
)

// TestBasicSDKUsageExample runs the curated README example end-to-end against
// the live Kestra container. This is the CI guarantee for the injected snippet:
// if flowLifecycle (and therefore the README block produced from it by
// test-utils/EmbedSnippets.java) ever stops matching the SDK, this test fails.
// See issue #144.
func TestBasicSDKUsageExample(t *testing.T) {
	ctx := context.Background()
	client := KestraTestClient()

	// Best-effort cleanup so the example is re-runnable within a session.
	defer func() {
		_ = client.Flows().DeleteFlow(ctx, "company.team", "hello_from_sdk", MAIN_TENANT)
	}()

	executionID, err := flowLifecycle(ctx, client, MAIN_TENANT)
	require.NoError(t, err)
	require.NotEmpty(t, executionID)
}
